lxd/db/cluster/identities.go

```diff
@@ -398,6 +398,30 @@ func ActivateTLSIdentity(ctx context.Context, tx *sql.Tx, identifier uuid.UUID,
 	return nil
 }
 
+var getPendingTLSIdentityByTokenSecretStmt = fmt.Sprintf(`
+SELECT identities.id, identities.auth_method, identities.type, identities.identifier, identities.name, identities.metadata
+	FROM identities
+	WHERE identities.type = %d
+	AND json_extract(identities.metadata, '$.secret') = ?
+`, identityTypeCertificateClientPending)
+
+// GetPendingTLSIdentityByTokenSecret gets a single identity of type identityTypeCertificateClientPending with the given
+// secret in its metadata. If no pending identity is found, an api.StatusError is returned with http.StatusNotFound.
+func GetPendingTLSIdentityByTokenSecret(ctx context.Context, tx *sql.Tx, secret string) (*Identity, error) {
+	identities, err := getIdentitysRaw(ctx, tx, getPendingTLSIdentityByTokenSecretStmt, secret)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(identities) == 0 {
+		return nil, api.NewStatusError(http.StatusNotFound, "No pending identities found with given secret")
+	} else if len(identities) > 1 {
+		return nil, errors.New("Multiple pending identities found with given secret")
```

Comment:
Do we check for conflicts when issuing tokens to prevent this scenario occurring?

Reply:
I'm using `shared.RandomCryptoString` to generate the secret as in `POST /1.0/certificates` which returns 32 random bytes. It's unlikely that there will be a collision, but we can check and regenerate?

Reply:
Probably fine for now then, and you've protected it anyway